chirho/counterfactual/handlers/counterfactual.py

```diff
@@ -100,23 +105,102 @@ class Preemptions(Generic[T], pyro.poutine.messenger.Messenger):
         or one of its subclasses, typically from an auxiliary discrete random variable.
 
     :param actions: A mapping from sample site names to interventions.
+    :param prefix: Prefix usable for naming any auxiliary random variables.
     """
 
     actions: Mapping[str, Intervention[T]]
+    prefix: str
 
-    def __init__(self, actions: Mapping[str, Intervention[T]]):
+    def __init__(
+        self, actions: Mapping[str, Intervention[T]], *, prefix: str = "__split_"
+    ):
         self.actions = actions
+        self.prefix = prefix
         super().__init__()
 
-    def _pyro_post_sample(self, msg: Dict[str, Any]) -> None:
```

Comment:
There's a bit of mismatch in the way `split` and `preempt` unpack their arguments versus `intervene` and the structure of the complex `Intervention` type in `chirho.interventional.ops`. I removed the annotation here to prevent `mypy` from causing CI failures over this minor issue, which can be dealt with directly in a separate PR.

Reply:
This issue is related to and perhaps subsumed by #26